unmarshal.go

```diff
@@ -242,6 +247,80 @@ func unmarshalEntries(v reflect.Value, entries []*Entry, opt *marshalOptions) er
 	return nil
 }
 
+func applyDefaultValue(rv reflect.Value, value string) error {
```

Comment:
I think this function can be eliminated?

Use `defaultValueFromTag()` to get a `*hcl.Value` then `unmarshalValue()` to apply it to the `reflect.Value`. Maybe wrap the error from `unmarshalValue` to provide some more useful context.

Reply:
well, that's the initial thought. `unmarshalValue`'s called when there's matching value between `.hcl` & `struct attribute`, doing what applyDefault there will basically does nothing.

Reply:
ended up removing the `applyDefaultValue` as suggested, thanks!!!